lib/libc/stdio/lib_tempnam.c

```diff
@@ -105,7 +105,16 @@ FAR char *tempnam(FAR const char *dir, FAR const char *pfx)
 	FAR char *path;
 	int ret;
 
-	(void)asprintf(&path, "%s/%s-XXXXXX.tmp", dir, pfx);
+	if (!pfx && !dir) {
+		return NULL;
```

Comment:
I think that it should be "asprintf(&path, "%s/XXXXXX.tmp", P_tmpdir);" in this case. 
So, I think it will be better as below.
	if (!dir) {
		dir = P_tmpdir;
	}

	if (!pfx) {
		(void)asprintf(&path, "%s/XXXXXX.tmp", dir);
	} else {
		(void)asprintf(&path, "%s/%s-XXXXXX.tmp", dir, pfx);
	}

How do you think about this?

Reply:
looks good to me